### PR TITLE
update status of clusterlogging using the status client

### PR DESCRIPTION
**Pull request by jcantrill**
Use the Status client since we should only update the status of a customresource

### CODE REVIEW

**Comment by jcantrill** [comment]
> Do we need to worry about getting the object before updating the status at all?

We should be the only one updating it but I don't know for certain.  @bparees or @chancez  do you know?

**Comment by bparees** [comment]
> Do we need to worry about getting the object before updating the status at all?

you mean to avoid an update conflict?  you should probably always handle potential update conflicts either explicitly (retry then and there) or by being confident that you'll get another watch event(due to the write that caused the conflict) which will drive the retry in the future.

File: pkg/k8shandler/clusterloggingrequest.go [hunk 1]
```diff
@@ -35,6 +35,15 @@ func (clusterRequest *ClusterLoggingRequest) Update(object runtime.Object) error
 	return clusterRequest.client.Update(context.TODO(), object)
 }
 
+//Update the runtime Object status or return error
+func (clusterRequest *ClusterLoggingRequest) UpdateStatus(object runtime.Object) (err error) {
+	logrus.Tracef("Updating Status: %v", object)
+	if err = clusterRequest.client.Status().Update(context.TODO(), object); err != nil {
```
[the remark this answers is not in the document]
Comment (jcantrill) [reply pkg/k8shandler/clusterloggingrequest.go]: I believe all the callers of this function handle conflicts

**Comment by richm** [comment]
@jcantrill the latest error does not appear to be a flake:
```
=== RUN   TestClusterLogging/collectors/fluentd
--- FAIL: TestClusterLogging (261.75s)
    --- FAIL: TestClusterLogging/collectors (257.41s)
        --- FAIL: TestClusterLogging/collectors/fluentd (197.41s)
        	clusterlogging_test.go:162: Initializing cluster resources...
...
        	wait_util.go:55: Waiting for availability of kibana deployment
        	clusterlogging_test.go:65: timed out waiting for the condition
```
I suggest you put this PR on hold until you are able to debug it, to avoid the noise from the retry bot.

**Comment by jcantrill** [comment]
:man_shrugging:   passes locally

**Comment by ewolinetz** [comment]
looks like kibana didn't start up... i would log into the ci console and poke around during a test run to see what is happening

**Comment by jcantrill** [comment]
I think Enabling status may fix this #298

**Pull request merged**